packages/editor-ui/src/components/ModalDrawer.vue

```diff
@@ -65,8 +68,22 @@ export default Vue.extend({
 				this.$emit('enter');
 			}
 		},
-		close() {
+		close(callback?: () => void) {
+			if (this.beforeClose) {
+				this.beforeClose(() => {
+					this.$store.commit('ui/closeTopModal');
+					if (typeof callback === 'function') {
+						callback();
+					}
+				});
+
+				return;
+			}
+
 			this.$store.commit('ui/closeTopModal');
+			if (typeof callback === 'function') {
+				callback();
+			}
 		},
```

Comment:
I am confused why you add the `callback` parameter if it's not getting used. And i am confused why you are passing callback to `beforeClose` if it's not getting used either.

Why not simply this?
```suggestion
		close() {
			if (this.onClose) {
			    this.onClose();
			}

			this.$store.commit('ui/closeTopModal');
		},
```

Reply:
Fixed